src/library_wasmfs_jsimpl.js

```diff
@@ -85,6 +80,7 @@ mergeInto(LibraryManager.library, {
     _emscripten_proxy_finish(ctx);
   },
 
+  _wasmfs_jsimpl_async_read__sig: 'vpppppjp',
```

Comment:
```suggestion
  _wasmfs_jsimpl_async_read__sig: 'vppppjp',
```

Reply:
That doesn't look right to me.. there are 6 pointer-width args before `offset` and one after it.   (also if I make this change then one of the tests fails).

Reply:
ah, sorry, I left this comment in the wrong place. fixing.